### PR TITLE
chore(kuma-init): use distroless image

### PR DESCRIPTION
**Pull request by lobkovilya**
Signed-off-by: Ilya Lobkov <ilya.lobkov@konghq.com>

### Checklist prior to review

- [ ] Link to docs PR or issue --
- [ ] Link to UI issue or PR --
- [ ] Is the [issue worked on linked][1]? --
- [ ] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [ ] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Unit Tests --
- [ ] E2E Tests --
- [ ] Manual Universal Tests --
- [ ] Manual Kubernetes Tests --
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [ ] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: tools/releases/dockerfiles/Dockerfile.kuma-init [hunk 1]
```diff
@@ -1,13 +1,8 @@
 # When updatinng image version, remember to update it in tools/releases/distros.sh and enterprise version
 FROM kumahq/kuma-net-ebpf:0.8.10 as ebpf
-# until there is a distroless iptables image we have to use something else
-FROM ubuntu:jammy
+FROM gcr.io/k8s-staging-build-image/distroless-iptables:v0.1.2
```
Comment (lahabana) [inline tools/releases/dockerfiles/Dockerfile.kuma-init:3]: Sounds ok but who maintains this atm?

Comment (lobkovilya) [reply tools/releases/dockerfiles/Dockerfile.kuma-init:3]: It's Kubernetes https://github.com/kubernetes/release/tree/master/images/build/distroless-iptables

**Comment by lobkovilya** [comment]
Random legacy tests are failing, I didn't have time to look into them. I should be able to do it later this week

**Comment by slonka** [comment]
I'm reviving this PR for https://github.com/kumahq/kuma/issues/11786 - we might not need apt-mirror if we don't have apt 
image

File: tools/releases/dockerfiles/kuma-init.Dockerfile [hunk 2]
```diff
@@ -1,21 +1,14 @@
-# until there is a distroless iptables image we have to use something else
-FROM ubuntu:jammy-20240808@sha256:adbb90115a21969d2fe6fa7f9af4253e16d45f8d4c1e930182610c4731962658
+FROM gcr.io/k8s-staging-build-image/distroless-iptables:v0.1.2
```
Comment (slonka) [inline tools/releases/dockerfiles/kuma-init.Dockerfile:1]:
```suggestion
FROM gcr.io/k8s-staging-build-image/distroless-iptables:v0.6.4
```

File: tools/releases/dockerfiles/kuma-init.Dockerfile [hunk 3]
```diff
@@ -1,21 +1,14 @@
...
 
 # this will be from a base image once it is done
 COPY /tools/releases/templates/LICENSE \
     /tools/releases/templates/README \
     /kuma/
 
-COPY /tools/releases/templates/NOTICE /kuma/NOTICE
-
-RUN adduser --system --disabled-password --group kumactl --uid 5678
+COPY /tools/releases/templates/NOTICE-kumactl /kuma/NOTICE
```
Comment (slonka) [inline tools/releases/dockerfiles/kuma-init.Dockerfile:11]:
```suggestion
COPY /tools/releases/templates/NOTICE /kuma/NOTICE
```

File: tools/releases/templates/group [hunk 4]
```diff
@@ -0,0 +1,6 @@
+root:x:0:
+nobody:x:65534:
+tty:x:5:
+staff:x:50:
+nonroot:x:65532:
```
Comment (slonka) [inline tools/releases/templates/group:5]: these lines are default in the distroless container, you  can check it by:

```
docker run --rm --entrypoint cat gcr.io/k8s-staging-build-image/distroless-iptables:v0.6.4 /etc/group
```

File: tools/releases/templates/passwd [hunk 5]
```diff
@@ -0,0 +1,4 @@
+root:x:0:0:root:/root:/sbin/nologin
+nobody:x:65534:65534:nobody:/nonexistent:/sbin/nologin
+nonroot:x:65532:65532:nonroot:/home/nonroot:/sbin/nologin
```
Comment (slonka) [inline tools/releases/templates/passwd:3]: same as above:

```
docker run --rm --entrypoint cat gcr.io/k8s-staging-build-image/distroless-iptables:v0.6.4 /etc/shadow
```

File: tools/releases/dockerfiles/kuma-init.Dockerfile [hunk 6]
```diff
@@ -1,11 +1,6 @@
-# until there is a distroless iptables image we have to use something else
-FROM ubuntu:jammy-20240808@sha256:adbb90115a21969d2fe6fa7f9af4253e16d45f8d4c1e930182610c4731962658
+FROM gcr.io/k8s-staging-build-image/distroless-iptables:v0.6.4
 ARG ARCH
 
-RUN apt-get update && \
```
Comment (jijiechen) [inline tools/releases/dockerfiles/kuma-init.Dockerfile:5]: So we don't need to install these packages anymore, I guess the CI issue which relies on `ubuntu-22.04` may also be resolved.

Comment (slonka) [reply tools/releases/dockerfiles/kuma-init.Dockerfile:5]: That could be another benefit, yeah.

**Review by jakubdyszkiewicz — APPROVED** [review]

**Comment by slonka** [comment]
closed via https://github.com/kumahq/kuma/pull/5945

**Pull request merged**